Tasks/AzureRmWebAppDeployment/operations/AzureAppServiceUtility.ts

```diff
@@ -0,0 +1,232 @@
+            }
+            else {
+                tl.debug(`Skipped updating the SCM value. Value: ${scmType}`);
+            }
+        }
+        catch(error) {
+            tl.warning(tl.loc("FailedToUpdateAzureRMWebAppConfigDetails", error));
+        }
+    }
+
+    public async getWebDeployPublishingProfile(): Promise<any> {
+        var publishingProfile = await this._appService.getPublishingProfileWithSecrets();
+        var defer = Q.defer<any>();
+        parseString(publishingProfile, (error, result) => {
+            for (var index in result.publishData.publishProfile) {
+                if (result.publishData.publishProfile[index].$.publishMethod === "MSDeploy") {
+                    defer.resolve(result.publishData.publishProfile[index].$);
+                }
+            }
+            defer.reject(tl.loc('ErrorNoSuchDeployingMethodExists'));
+        });
+
+        return defer.promise;
+    }
+
+    public async getApplicationURL(): Promise<string> {
+        let webDeployProfile: any =  await this.getWebDeployPublishingProfile();
+        return await webDeployProfile.destinationAppUrl;
+    }
+
+    public async pingApplication(): Promise<void> {
+        try {
+            var applicationUrl: string = (await this.getWebDeployPublishingProfile()).destinationAppUrl;
+
+            if(!applicationUrl) {
+                tl.debug('Application Url not found.');
+                return;
+            }
+            var webRequest = new webClient.WebRequest();
+            webRequest.method = 'GET';
+            webRequest.uri = applicationUrl;
+            tl.debug('pausing for 5 seconds before request');
+            await webClient.sleepFor(5);
+            var response = await webClient.sendRequest(webRequest);
+            tl.debug(`App Service status Code: '${response.statusCode}'. Status Message: '${response.statusMessage}'`);
+        }
+        catch(error) {
+            tl.debug(`Unable to ping App Service. Error: ${error}`);
+        }
+    }
+
+    public async getKuduService(): Promise<Kudu> {
+        var publishingCredentials = await this._appService.getPublishingCredentials();
+        if(publishingCredentials.properties["scmUri"]) {
+            tl.setVariable(`AZURE_APP_SERVICE_KUDU_${this._appService.getSlot()}_PASSWORD`, publishingCredentials.properties["publishingPassword"], true);
+            return new Kudu(publishingCredentials.properties["scmUri"], publishingCredentials.properties["publishingUserName"], publishingCredentials.properties["publishingPassword"]);
+        }
+
+        throw Error(tl.loc('KuduSCMDetailsAreEmpty'));
+    }
+
+    public async getPhysicalPath(virtualApplication: string): Promise<string> {
+
+        if(!virtualApplication) {
+            return '/site/wwwroot';
+        }
+
+        virtualApplication = (virtualApplication.startsWith("/")) ? virtualApplication.substr(1) : virtualApplication;
+
+        var appConfigSettings = await this._appService.getConfiguration();
```

Comment:
Should this line be inside _getPhysicalToVirtualPathMap, will discuss offline.

Reply:
+1